wkt_to_dicomseg/main.ipynb

```diff
@@ -0,0 +1,287 @@
+    "                )\n",
+    "                segment_descriptions.append(description)\n",
+    "            else:\n",
+    "                print(f\"No valid mask created for geometry {idx}\")\n",
+    "\n",
+    "        if not all_masks:\n",
+    "            raise ValueError(\"No valid masks were created from the geometry data\")\n",
+    "\n",
+    "        masks_array = np.stack(all_masks, axis=0)\n",
+    "        \n",
+    "        visualize_dicom_and_masks(dicom_file, masks_array)\n",
+    "\n",
+    "        seg_dataset = hd.seg.Segmentation(\n",
+    "            source_images=[dicom_dataset],\n",
+    "            pixel_array=masks_array,\n",
+    "            segmentation_type=hd.seg.SegmentationTypeValues.BINARY,\n",
+    "            segment_descriptions=segment_descriptions,\n",
+    "            series_instance_uid=hd.UID(),\n",
+    "            series_number=1,\n",
+    "            sop_instance_uid=hd.UID(),\n",
+    "            instance_number=1,\n",
+    "            manufacturer='Manufacturer',\n",
+    "            manufacturer_model_name='Model',\n",
+    "            software_versions='v1.0',\n",
+    "            device_serial_number='0000'\n",
+    "        )\n",
+    "\n",
+    "        # Save the DICOM SEG file\n",
+    "        seg_dataset.save_as(output_path)\n",
+    "        return {\"message\": \"DICOM SEG file created successfully\", \"file_path\": output_path}\n",
+    "\n",
+    "    except Exception as e:\n",
+    "        raise  # Re-raise the exception after logging it"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "6edfe0ef-9847-430d-84e9-c21d20bc1fc3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def visualize_dicom_and_masks(dicom_path, masks_array):\n",
+    "    \"\"\"\n",
+    "    Visualize DICOM image with mask overlays.\n",
+    "    \n",
+    "    Args:\n",
+    "        dicom_path: Path to the DICOM file\n",
+    "        masks_array: 3D numpy array of masks (num_masks, height, width)\n",
+    "    \"\"\"\n",
+    "    # Read DICOM file\n",
+    "    dicom_dataset = pydicom.dcmread(dicom_path)\n",
+    "    dicom_image = dicom_dataset.pixel_array\n",
+    "    \n",
+    "    # Calculate number of subplots needed\n",
+    "    num_masks = masks_array.shape[0]\n",
+    "    total_plots = num_masks + 1  # +1 for original image\n",
+    "    \n",
+    "    # Create figure with subplots\n",
+    "    fig, axes = plt.subplots(1, total_plots, figsize=(5*total_plots, 5))\n",
+    "    if total_plots == 2:  # Handle case of single mask\n",
+    "        axes = [axes[0], axes[1]]\n",
+    "    \n",
+    "    # Plot original DICOM image\n",
+    "    axes[0].imshow(dicom_image, cmap='gray')\n",
+    "    axes[0].set_title('Original DICOM')\n",
+    "    axes[0].axis('off')\n",
+    "    \n",
+    "    # Plot each mask as overlay\n",
+    "    colors = ['red', 'green', 'blue']  # Different color for each mask\n",
```

Comment:
It doesn't look like this is getting used. Also, would it make more sense to map color based on intensity or answer class?